src/main/java/io/cryostat/core/agent/AgentJMXHelper.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.core.agent;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.management.MBeanServerConnection;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+import org.openjdk.jmc.rjmx.ConnectionException;
+import org.openjdk.jmc.rjmx.IConnectionHandle;
+
+public class AgentJMXHelper {
+
+    private static final Logger logger = Logger.getLogger(AgentJMXHelper.class.getName());
+    private static final String AGENT_OBJECT_NAME =
+            "org.openjdk.jmc.jfr.agent:type=AgentController";
+    private static final String DEFINE_EVENT_PROBES = "defineEventProbes";
+    private static final String RETRIEVE_EVENT_PROBES = "retrieveEventProbes";
+    private static final String RETRIEVE_CURRENT_TRANSFORMS = "retrieveCurrentTransforms";
+
+    private final IConnectionHandle connectionHandle;
+    private final MBeanServerConnection mbsc;
+
+    public AgentJMXHelper(IConnectionHandle connectionHandle) throws ConnectionException {
+        this.connectionHandle = connectionHandle;
+        mbsc = connectionHandle.getServiceOrDummy(MBeanServerConnection.class);
+    }
+
+    public IConnectionHandle getConnectionHandle() {
+        return connectionHandle;
+    }
+
+    public MBeanServerConnection getMBeanServerConnection() {
+        return mbsc;
+    }
+
+    public boolean isMXBeanRegistered() {
+        try {
+            return mbsc.isRegistered(new ObjectName(AGENT_OBJECT_NAME));
+        } catch (MalformedObjectNameException | IOException e) {
+            logger.log(Level.SEVERE, "Could not check if agent MXBean is registered", e);
+        }
+        return false;
+    }
+
+    public String retrieveEventProbes() {
+        try {
+            Object result =
+                    mbsc.invoke(
+                            new ObjectName(AGENT_OBJECT_NAME),
+                            RETRIEVE_EVENT_PROBES,
+                            new Object[0],
+                            new String[0]);
+
+            return result.toString();
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "Could not retrieve event probes", e);
+        }
+        return null;
```

Comment:
I'd rather not return nulls if we can avoid it. Could those caught exceptions be thrown instead of logged and swallowed? Or, could the return type be `Optional`?

Reply:
I've reworked this to rethrow the exception, thanks